include/faabric/util/dirty.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
```

Comment:
`dirty.h` and `dirty.cpp` contain all the boilerplate code around dirty page tracking using both soft PTEs and segfaults.